From: Mike Kasick <mike@kasick.org>
Date: Tue, 22 Nov 2011 17:10:21 -0500

Make "reboot recovery" use bml8 recovery (fota.rc).

Also add "reboot bml7recovery" to boot bml7 recovery (recovery.rc).

diff --git a/kernel/arch/arm/mach-s5pv210/mach-victory.c b/kernel/arch/arm/mach-s5pv210/mach-victory.c
--- a/kernel/arch/arm/mach-s5pv210/mach-victory.c
+++ b/kernel/arch/arm/mach-s5pv210/mach-victory.c
@@ -153,11 +153,11 @@
 	int mode = REBOOT_MODE_NONE;
 
 	if ((code == SYS_RESTART) && _cmd) {
-		if (!strcmp((char *)_cmd, "arm11_fota"))
+		if (!strcmp((char *)_cmd, "arm11_fota") || !strcmp((char *)_cmd, "recovery"))
 			mode = REBOOT_MODE_ARM11_FOTA;
 		else if (!strcmp((char *)_cmd, "arm9_fota"))
 			mode = REBOOT_MODE_ARM9_FOTA;
-		else if (!strcmp((char *)_cmd, "recovery"))
+		else if (!strcmp((char *)_cmd, "bml7recovery"))
 			mode = REBOOT_MODE_RECOVERY;
 		else if (!strcmp((char *)_cmd, "bootloader"))
 			mode = REBOOT_MODE_FAST_BOOT;
